riscv/csrs.cc

```diff
@@ -191,7 +191,7 @@ bool pmpaddr_csr_t::access_ok(access_type type, reg_t mode) const noexcept {
   const bool typer = type == LOAD;
   const bool typex = type == FETCH;
   const bool typew = type == STORE;
-  const bool normal_rwx = (typer && cfgr) || (typew && cfgw) || (typex && cfgx);
+  const bool normal_rwx = ((typer && cfgr) && (!hlvx || cfgx)) || (typew && cfgw) || (typex && cfgx);
```

Comment:
Minor detail, but omitting the parentheses around `typer && cfgr`:

```
(typer && cfgr && (!hlvx || cfgx)) || (typew && cfgw) || (typex && cfgx)
```

is slightly clearer to me. Otherwise this seems to be right, HLVX requires both R and X PMP attributes.

Maybe we should add some assertion that `hlvx` implies `typer`? Or at least a comment on it.